### PR TITLE
config: Remove already deprecated `store.max-look-back-period`.

### PR DESCRIPTION
**Pull request by kavirajk**
**What this PR does / why we need it**:
This is already been deprecated longback. Decided to remove in next versions of Loki

**Which issue(s) this PR fixes**:
Fixes #NA

**Special notes for your reviewer**:
Checked if we use it anywhere in jsonnet and internal deployment. Couldn't find any reference. 

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory.

### CODE REVIEW

**Review by MichelHollands — APPROVED** [review]
LGTM

File: pkg/storage/config/store.go [hunk 1]
```diff
@@ -47,14 +41,8 @@ func (cfg *ChunkStoreConfig) RegisterFlags(f *flag.FlagSet) {
 	cfg.WriteDedupeCacheConfig.RegisterFlagsWithPrefix("store.index-cache-write.", "", f)
 
 	f.Var(&cfg.CacheLookupsOlderThan, "store.cache-lookups-older-than", "Cache index entries older than this period. 0 to disable.")
-	f.Var(&cfg.MaxLookBackPeriod, "store.max-look-back-period", "This flag is deprecated. Use -querier.max-query-lookback instead.")
 }
 
 func (cfg *ChunkStoreConfig) Validate(logger log.Logger) error {
-	if cfg.MaxLookBackPeriod > 0 {
```
Comment (ashwanthgoli) [inline pkg/storage/config/store.go:54]: nit: remove this method and it's usages?

**Review by ashwanthgoli — COMMENTED** [review]
PR checklist: this also needs to be added to the deletion list
```
 If the change is deprecating or removing a configuration option, update the deprecated-config.yaml and deleted-config.yaml files respectively in the tools/deprecated-config-checker directory. 
```

File: pkg/storage/config/store.go (hunk 1 above)
Comment (kavirajk) [reply pkg/storage/config/store.go:54]: thought about that. But left it as it as to be consistent with other configs. 

We check all the configs validate [here](https://github.com/grafana/loki/blob/kavirajk%2Fremove-store-look-back-period/pkg/loki/loki.go#L247-L249)

**Pull request merged**